### DISCUSSION TITLE
General error on live environment Laravel

### DISCUSSION THREAD

**Discussion by ruben-bolink** [spatie/laravel-pdf#93] [Q&A]
When running this local, it works perfect (converting tailwind css to pdf). On live environment (Linux, OS Debian buster 10) it gives an error.

 I tried adjusting the NPM (which npm = /usr/bin/npm) and node (which node = /usr/bin/node) paths. This gives the same error.

Code:
`return pdf()->view('quote_pdf', ['quoteData' => $quoteData, 'productRows' => $productRows, 'billingAddress' => $currentBillingAddress, 'endDate' => $endDate, 'vat' => $vat])
                ->withBrowsershot(function (Browsershot $browsershot) {
                    $browsershot->newHeadless()->setIncludePath('$PATH:/usr/bin');
                })
                ->name('#'.$quote->id.' - '.$quote->customer->company_name.' - '.$quote->customer->last_name);`

Error:
`The command "PATH=$PATH:/usr/bin NODE_PATH=`npm root -g` node '/data/web/apps/quotes.maxiaxi.com/releases/87/vendor/spatie/browsershot/src/../bin/browser.cjs' '{"url":"file:\/\/\/tmp\/1932751629-0168181001709298436\/index.html","action":"pdf","options":{"args":[],"viewport":{"width":800,"height":600},"displayHeaderFooter":false,"newHeadless":true,"printBackground":true}}'" failed.

Exit Code: 1(General error)

Working directory: /data/web/apps/quotes.maxiaxi.com/releases/87/public

Output:
================

Error Output:
================
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
Error: Failed to launch the browser process!
/data/web/.cache/puppeteer/chrome/linux-122.0.6261.69/chrome-linux64/chrome: error while loading shared libraries: libgbm.so.1: cannot open shared object file: No such file or directory

TROUBLESHOOTING: https://pptr.dev/troubleshooting

at Interface.onClose (/data/web/apps/quotes.maxiaxi.com/releases/87/node_modules/@puppeteer/browsers/lib/cjs/launch.js:267:24)
at Interface.emit (node:events:539:35)
at Interface.close (node:internal/readline/interface:535:10)
at Socket.onend (node:internal/readline/interface:261:10)
at Socket.emit (node:events:539:35)
at endReadableNT (node:internal/streams/readable:1344:12)
at process.processTicksAndRejections (node:internal/process/task_queues:82:21)`

Does anyone know a solution for this? thank you in advance

**Comment by dylanBcp** [comment]
I have the same error on live server

**Comment by albertStaalburger** [comment]
Facing a similar issue here. 
Environment: `Distributor ID: Ubuntu ,Description: Ubuntu 20.04.6 LTS, Release: 20.04, Codename: focal`
Kernel: `Linux 5.4.0-176-generic x86_64`

I have puppeteer installed and the `~/.cache/puppeteer` exists. I used `ldd chrome | grep not` to find and install the missing packages, however I still get the general error `Error: Could not find Chrome (ver. 123.0.6312.105).`
I then tried solution 1 `npx puppeteer browsers install chrome` which installed the exact version the  error reported was missing

`chrome@123.0.6312.105 /home/xxxx/.cache/puppeteer/chrome/linux-123.0.6312.105/chrome-linux64/chrome`

This still resulted in the same error. I then tried solution 2 `your cache path is incorrectly configured (which is: /var/www/.cache/puppeteer)` which obviously differs from the installation path which is at `$HOME/.cache/puppeteer`
So I used the suggested code snippet to set the puppeteer cache dir, and then ran `npx puppeteer browsers install chrome` becuse the dir did not yet exist. This created the cache dir with the required version of chrome installed.

This time the error changed to :
`Error: Failed to launch the browser process! chrome_crashpad_handler: --database is required Try 'chrome_crashpad_handler --help' for more information. [29215:29215:0411/091416.581593:ERROR:socket.cc(120)] recvmsg: Connection reset by peer (104) TROUBLESHOOTING: https://pptr.dev/troubleshooting at Interface.onClose`

I tried ldd again an no packages were reported as missing, the cache dir has been set correctly and I am none the wiser how to solve this. Anybody else mayby having the sam issue or have been able to fix it? Is this a permission or a configuration/installation issue?

**Reply by albertStaalburger** [reply]
Installing Puppeteer globally gave a completely different bag of issues!!!
